src/sources/statsd/parser.rs

```diff
@@ -135,11 +137,8 @@ fn parse_tags(input: &&str) -> Result<MetricTags, ParseError> {
         .map(|chunk| {
             let pair: Vec<_> = chunk.split(':').collect();
             let key = &pair[0];
-            // same as in telegraf plugin:
-            // if tag value is not provided, use "true"
-            // https://github.com/influxdata/telegraf/blob/master/plugins/inputs/statsd/datadog.go#L152
-            let value = pair.get(1).unwrap_or(&"true");
-            ((*key).to_owned(), (*value).to_owned())
+            let tag_value: TagValue = pair.get(1).map(|s| String::from(s.to_owned())).into();
+            ((*key).to_owned(), tag_value)
```

Comment:
This has a few issues we may want to address while updating this support. What I'm seeing:
1. If `chunk` contains more than one `:`, it will ignore everything after the second `:`.
2. If we don't want to ignore the above, we could split with `let colon = chunk.find(':');` instead of allocating an array.

Reply:
> Also, as this is a breaking change, should this add to the release highlights doc now?

Yep, That was flagged as a TODO in the opening comment.

Reply:
> If chunk contains more than one :, it will ignore everything after the second :

I'm no statsd expert. Like, I literally used it for the first time working on this diff.

But I tried to get statsd to include a colon in one of the values, and it doesn't seem to allow that.
I think it takes the first colon it sees and then removes all after it (up until the next `,`).

I can add a comment describing that is the behavior.

Input:

```
foo:1|c|#tag3:foo:bar" | nc -u -w0 127.0.0.1 8125
```

output (the colon was removed):

```
'foo;tag3=foobar'
```

input:

```
echo "foo:1|c|#tag3:foo::bar" | nc -u -w0 127.0.0.1 8125
```

output (both removed):

```
'foo;tag3=foobar'
```

input:

```
echo "foo:1|c|#tag3:foo:::bar" | nc -u -w0 127.0.0.1 8125
```

output:

output (all removed):

```
'foo;tag3=foobar'
```

Reply:
Yes, that's worth highlighting in this section, and probably in the docs as well. I still think this would be better solved with a `find` rather than splitting into a new vec, but that is orthogonal to your change and could be cleaned up later.

Reply:
For our less savvy reviewers (it me), those are counters named "foo" with a value of "1" and intending tags to be `"tag3" = "foo:bar|foo::bar|foo:::bar"`?

Reply:
>  I still think this would be better solved with a find rather than splitting into a new vec

I'll update it to use `find()` :+1:  , I'm using that in the `datadog_agent` changes.

> For our less savvy reviewers 

Yes!

Reply:
addressed in https://github.com/vectordotdev/vector/pull/15617/commits/f106fe7bdb9aedcf5a64228ef4d629afccb3077e

Reply:
doh, still need to add a comment / docs update